src/jit/simd.cpp

```diff
@@ -2349,11 +2349,14 @@ GenTree* Compiler::impSIMDIntrinsic(OPCODE                opcode,
                                     CORINFO_CLASS_HANDLE  clsHnd,
                                     CORINFO_METHOD_HANDLE methodHnd,
                                     CORINFO_SIG_INFO*     sig,
+                                    unsigned              methodFlags,
                                     int                   memberRef)
 {
     assert(featureSIMD);
 
-    if (!isSIMDClass(clsHnd))
+    // Exit early if we are either not in one of the SIMD types or if the method
+    // is not a JIT Intrinsic (which requires the [Intrinsic] attribute).
+    if (!isSIMDClass(clsHnd) || ((methodFlags & CORINFO_FLG_JIT_INTRINSIC) == 0))
```

Comment:
When validating locally, I caught all the methods that we were implicitly treating as intrinsic (all the ones above that I added the `[Intrinsic]` attribute to) by adding an assert that `(methodFlags & CORINFO_FLG_JIT_INTRINSIC) != 0` on L2363 (so after the `getSIMDIntrinsicInfo` call and check that the intrinsic returned was valid).

I then generated PMI diffs for `-c` (corelib), `-f` (framework), `--benchmarks`, and `--tests`; which hit the assert for everything that was being implicitly treated as `[Intrinsic]`.

Reply:
Did you get to zero diffs once everything was properly marked? It seems that you decided to make the change to `Dot`, so presumably that would lead to diffs?

Reply:
Yes, I got to zero-diffs once everything was properly marked.

With the change to `Dot` and `AndNot` (making them recognized), there are now diffs for those call-sites. I'll collect them again and post.

Reply:
@CarolEidt, I reran the diffs (`-c`, `-f`, `--benchmarks`, and `--tests`) using the current HEAD as base and this PR + the CoreFX side changes (https://github.com/dotnet/corefx/pull/34167) as the diff.

The only delta was in `S.P.Corelib` (due to `Vector.AndNot` now being treated as an intrinsic):
```
Found 1 files with textual diffs.
PMI Diffs for System.Private.CoreLib.dll for x64 default jit
Summary:
(Lower is better)
Total bytes of diff: -402 (-0.01% of base)
    diff is an improvement.
Top file improvements by size (bytes):
        -402 : System.Private.CoreLib.dasm (-0.01% of base)
1 total files with size differences (1 improved, 0 regressed), 0 unchanged.
Top method improvements by size (bytes):
        -402 (-41.40% of base) : System.Private.CoreLib.dasm - Vector`1:ConditionalSelect(struct,struct,struct):struct (4 methods)
Top method improvements by size (percentage):
        -402 (-41.40% of base) : System.Private.CoreLib.dasm - Vector`1:ConditionalSelect(struct,struct,struct):struct (4 methods)
1 total methods with size differences (1 improved, 0 regressed), 14260 unchanged.
Completed analysis in 3.13s
```

I can share the full delta if you're interested, but it basically just comes down to replacing the following large sequence of instructions:
```asm
       mov      rcx, 0xD1FFAB1E
       mov      edx, 252
       vextractf128 ymm8, ymm6, 1
       vextractf128 ymm9, ymm7, 1
       call     CORINFO_HELP_CLASSINIT_SHARED_DYNAMICCLASS
       vinsertf128 ymm7, ymm9, 1
       vinsertf128 ymm6, ymm8, 1
       mov      rax, 0xD1FFAB1E
       mov      rax, gword ptr [rax]
       vmovupd  ymm0, ymmword ptr[rax+8]
       vxorpd   ymm0, ymm7
       vandpd   ymm0, ymm6, ymm0
       vmovupd  ymm1, ymmword ptr[rdi]
       vmovupd  ymm2, ymmword ptr[rsi]
       vandpd   ymm1, ymm2
```

with:
```asm
vandnpd  ymm1, ymm2
```

Reply:
I also explicitly got diffs for `System.Numerics.Vectors.PerformanceTests`, which had no diffs.


and for `System.Numerics.Vectors.Tests`, which had a couple:

```
Found 1 files with textual diffs.
PMI Diffs for E:\Users\tagoo\Repos\corefx\artifacts\bin\System.Numerics.Vectors.Tests\netcoreapp-Release\System.Numerics.Vectors.Tests.dll for x64 default jit
Summary:
(Lower is better)
Total bytes of diff: -48 (-0.03% of base)
    diff is an improvement.
Top file improvements by size (bytes):
         -48 : System.Numerics.Vectors.Tests.dasm (-0.03% of base)
1 total files with size differences (1 improved, 0 regressed), 0 unchanged.
Top method improvements by size (bytes):
         -48 (-2.14% of base) : System.Numerics.Vectors.Tests.dasm - GenericVectorTests:TestBitwiseAndNot():this (4 methods)
Top method improvements by size (percentage):
         -48 (-2.14% of base) : System.Numerics.Vectors.Tests.dasm - GenericVectorTests:TestBitwiseAndNot():this (4 methods)
1 total methods with size differences (1 improved, 0 regressed), 980 unchanged.
Completed analysis in 0.45s
```

Reply:
> I can share the full delta if you're interested

No need - just wanted to make sure someone had a look.